src/baldr/datetime.cc

```diff
@@ -13,28 +13,59 @@
 #include "midgard/logging.h"
 #include "midgard/util.h"
 
+namespace {
+// use a cache to store already constructed sys_info's since they aren't cheap
+template <typename TP>
+const date::sys_info&
+from_cache(const TP& tp,
+           const date::time_zone* tz,
+           std::unordered_map<const date::time_zone*, std::vector<date::sys_info>>& cache) {
+  // check if we have anything for this timezone in the cache
+  auto tz_it = cache.find(tz);
```

Comment:
Is it ok to use a pointer as the cache key? Is each timezone only instantiated once?

Reply:
They are statically initialized in a singleton